File: automation/app_controller.py
"""Cross-platform app launcher utility."""

import os
import platform
import subprocess


def open_app(app_name: str) -> dict:
    system = platform.system().lower()
    if system == "windows":
        os.startfile(app_name)  # type: ignore[attr-defined]
    elif system == "darwin":
        subprocess.Popen(["open", "-a", app_name])
    else:
        subprocess.Popen(["xdg-open", app_name])
    return {"success": True, "app": app_name}

File: automation/app_controller.js
const { execFile } = require("child_process");

function openApp(appName) {
  return new Promise((resolve, reject) => {
    const platform = process.platform;
    let cmd;
    let args;

    if (platform === "win32") {
      cmd = "cmd";
      args = ["/c", "start", "", appName];
    } else if (platform === "darwin") {
      cmd = "open";
      args = ["-a", appName];
    } else {
      cmd = "xdg-open";
      args = [appName];
    }

    execFile(cmd, args, (error) => {
      if (error) {
        reject(error);
        return;
      }
      resolve({ success: true, appName });
    });
  });
}

module.exports = { openApp };
